### PR TITLE
fix: extra space in post text after urls

### PR DESCRIPTION
**Pull request by ice-endymion**
## Description
Add Post-process to handle spaces in text matches after URLs

## Type of Change
- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Refactoring
- [ ] Documentation
- [ ] Chore

## Screenshots

<table>
  <tr>
    <td><b>Before</b></td>
    <td><b>After</b></td>
  </tr>
  <tr>
    <td>Before</td>
    <td>After</td>
  </tr>
</table>

### CODE REVIEW

File: lib/app/features/feed/views/components/post/components/post_body/post_body.dart [hunk 1]
```diff
@@ -48,4 +54,24 @@ class PostBody extends HookConsumerWidget {
...
+  // Post-process text span to remove leading space
+  TextSpan _postProcessTextSpan(TextSpan postText) {
+    final children = postText.children;
+    if (children == null || children.isEmpty) return postText;
+
+    final firstSpan = children.first;
+    if (firstSpan is! TextSpan) return postText;
+
+    final text = firstSpan.text;
+    if (text == null || !text.startsWith(' ')) return postText;
```
[the remark this answers is not in the document]
Comment (ice-endymion) [reply lib/app/features/feed/views/components/post/components/post_body/post_body.dart:67]: So the source of problem is in how we build `content` string, so if we add media we have such edge case.
At first I fixed it in [parser](https://github.com/ice-blockchain/ion-app/pull/552/commits/fa9203e027bdd17acd3fe7b45cb8750a21b53374), but then unit test failed, cuz we also need to support regular URLs.

**Review by ice-kreios — APPROVED** [review]

**Review by ice-alcides — APPROVED** [review]
